cmake/thirdparty/SetupHIP.cmake

```diff
@@ -15,24 +15,76 @@ find_package(HIP REQUIRED)
 message(STATUS "HIP version:      ${HIP_VERSION_STRING}")
 message(STATUS "HIP platform:     ${HIP_PLATFORM}")
 
-if("${HIP_PLATFORM}" STREQUAL "hcc" OR "${HIP_PLATFORM}" STREQUAL "amd")
-	set(HIP_RUNTIME_DEFINE "__HIP_PLATFORM_HCC__")
-elseif("${HIP_PLATFORM}" STREQUAL "nvcc")
-	set(HIP_RUNTIME_DEFINE "__HIP_PLATFORM_NVCC__")
+set(HIP_RUNTIME_INCLUDE_DIRS "${HIP_ROOT_DIR}/include")
+if(${HIP_PLATFORM} STREQUAL "hcc")
+	set(HIP_RUNTIME_DEFINES "-D__HIP_PLATFORM_HCC__")
+    set(HIP_RUNTIME_LIBRARIES "${HIP_ROOT_DIR}/lib/libhip_hcc.so")
+elseif(${HIP_PLATFORM} STREQUAL "clang" OR ${HIP_PLATFORM} STREQUAL "amd")
+    set(HIP_RUNTIME_DEFINES "-D__HIP_PLATFORM_HCC__;-D__HIP_ROCclr__")
+    set(HIP_RUNTIME_LIBRARIES "${HIP_ROOT_DIR}/lib/libamdhip64.so")
+elseif(${HIP_PLATFORM} STREQUAL "nvcc")
+    set(HIP_RUNTIME_DEFINES "-D__HIP_PLATFORM_NVCC__")
+    find_package(cudatoolkit)
```

Comment:
CUDAToolKit is only available as of CMake 3.17, so this should be guarded by a version check (with either a fatal error or alternative way of locating the runtime library for older versions)

Reply:
Version check added and test that both routes with with CMake 3.11.4 and 3.18.2